src/libraries/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -528,6 +528,64 @@ public SequencePosition GetPosition(long offset)
             return Seek(offset);
         }
 
+        /// <summary>
+        /// Returns the offset of a <paramref name="position" /> within this sequence from the start.
+        /// </summary>
+        /// <param name="position">The <see cref="System.SequencePosition"/> of which to get the offset.</param>
+        /// <returns>The offset from the start of the sequence.</returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">The provided <see cref="System.SequencePosition"/> is invalid.</exception>
```

Comment:
It seems we would throw `ArgumentOutOfRangeException` when the position is out of range in 3 cases: the two you added, plus `BoundsCheck`:
https://source.dot.net/#System.Memory/System/Buffers/ReadOnlySequence.Helpers.cs,a8aafc415e4587c1

Should the exception message indicate that instead?:

```suggestion
        /// <exception cref="System.ArgumentOutOfRangeException">The position is out of range.</exception>
```

Reply:
I think so, we could add `The provided <see cref="System.SequencePosition"/> is out of range.`
BoundCheck throws only out of range exc if I'm not mistaken.